test/integration/services/data/test-get-claim-list-for-search.js

```diff
@@ -0,0 +1,195 @@
+const config = require('../../../../knexfile').intweb
+const knex = require('knex')(config)
+
+var expect = require('chai').expect
+var dateFormatter = require('../../../../app/services/date-formatter')
+var databaseHelper = require('../../../helpers/database-setup-for-tests')
+
+var testData
+var reference1 = 'SEARCH1'
+var reference2 = 'SEARCH2'
+var reference3 = 'ASDFGH1'
+
+var testFirstName = 'John'
+var testLastName = 'Smith'
+var name = testFirstName + ' ' + testLastName
+
+var date
+var getClaimsListForSearch = require('../../../../app/services/data/get-claim-list-for-search')
+
+describe('services/data/get-claim-list-for-search', function () {
+  this.timeout(10000)
+  before(function () {
+    date = dateFormatter.now()
+    testData = databaseHelper.getTestData(reference1, 'TESTING')
+
+    return databaseHelper.insertTestData(reference1, date.toDate(), 'TESTING')
+      .then(function () {
+        return databaseHelper.insertTestData(reference2, date.toDate(), 'TESTING')
+          .then(function () {
+            return databaseHelper.insertTestData(reference3, date.toDate(), 'TESTING')
+              .then(function () {
+                return knex('Visitor')
+                  .update({
+                    'NationalInsuranceNumber': '00000',
+                    'FirstName': 'Ref3FirstName',
+                    'LastName': 'Ref3LastName'
+                  })
+                  .where('Reference', reference3)
+                  .then(function () {
+                    return knex('Prisoner')
+                      .update({
+                        'PrisonNumber': '00000'
+                      })
+                      .where('Reference', reference3)
+                  })
+              })
+          })
+      })
```

Comment:
Refactor the above so that a Promise.all is returned.  This should improve performance and reduce the number of timeouts.  Give me a shout if you need